Updated moveit2_tutorials cmake

MarqRazz opened the pull request:
### Description

This updates the base cmake file to build with colcon. All of the `add_subdirectory` have been commented out and to enable the package you want to port you just uncomment it. I tried to use `COLCON_IGNORE` but because we are using `add_subdirectory` colcon would not see it and still attempted to build it.

### Checklist
- [x] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [x] While waiting for someone to review your request, please consider reviewing [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

MarqRazz commented on CMakeLists.txt:
```diff
@@ -1,75 +1,85 @@
-    interactive_markers
-    tf2_geometry_msgs
-  DEPENDS
-    EIGEN3
```
I'm not sure I didn't miss something here with EIGEN.

MarqRazz commented on CMakeLists.txt:
```diff
@@ -1,75 +1,85 @@
+find_package(tf2_geometry_msgs REQUIRED)
 
 set(THIS_PACKAGE_INCLUDE_DIRS
+  ament_cmake
```
Sorry I am not the best at cmake and was using the move_group.cmake to help me port this. Any help here would be appreciated.

probably need to remove the following from this list:
`std_srvs`
`moveit_ros_occupancy_map_monitor`
`moveit_ros_planning`
`pluginlib`

MarqRazz commented:
So I think this is ready to merge @henningkayser and @auman66. 

When the user to ready to port a package they will need to uncomment the desired subdirectory they are working in (i.e. `add_subdirectory(doc/moveit_cpp)`) and then port that packages `CMakeLists.txt` / source code.

Here is an example of what I think`doc/moveit_cpp/CMakeLists.txt` will look like after porting...
```
add_executable(moveit_cpp_tutorial src/moveit_cpp_tutorial.cpp)
target_include_directories(moveit_cpp_tutorial PUBLIC include)
ament_target_dependencies(moveit_cpp_tutorial  ${THIS_PACKAGE_INCLUDE_DEPENDS} Boost)

install(TARGETS moveit_cpp_tutorial 
  DESTINATION lib/${PROJECT_NAME}
)
install(DIRECTORY launch
  DESTINATION share/${PROJECT_NAME}
)
install(DIRECTORY config
  DESTINATION share/${PROJECT_NAME}
)
```

MarqRazz commented:
@henningkayser you asked in the standup if I had built the html pages locally but it looks like someone will need to port [rosdoc_lite](https://github.com/ros-infrastructure/rosdoc_lite) to ros2 before we can do that.

MarqRazz commented on CMakeLists.txt:
```diff
@@ -1,75 +1,85 @@
 
-add_subdirectory(doc/controller_configuration)
+# add_subdirectory(doc/controller_configuration)
 add_subdirectory(doc/hand_eye_calibration)
```
Should we remove the empty CMakeLists.txt files in this package? The only reason this one is uncommitted is because it has an cmake file.
[hand_eye_calibration](https://github.com/ros-planning/moveit2_tutorials/blob/main/doc/hand_eye_calibration/CMakeLists.txt)
[planning](https://github.com/ros-planning/moveit2_tutorials/blob/main/doc/planning/CMakeLists.txt)

henningkayser replied:
I guess we should remove all empty cmake files, don't know why we would have them in the first place

auman66 commented:
Any idea why the 2nd CI check is failing? 

The 3rd & 4th are a known issue with Docker and will hopefully be settled at some point (issue with docker throttling)

MarqRazz commented:
> @MarqRazz - Any idea why the 2nd CI check is failing?
> 
> The 3rd & 4th are a known issue with Docker and will hopefully be settled at some point (issue with docker throttling)

@auman66 it is failing because the package `ament_cmake_gtest` is not released for noetic. Currently there is no testing enabled but if I remove that package from the package.xml then if fails on `ament_cmake` which is also not released for noetic. Even if we could get the packages the next part travis would fail on is building the workspace because you cant run `catkin build` on the updated cmake file...
![image](https://user-images.githubusercontent.com/25058794/108221002-8e7fbf80-70f4-11eb-9b3e-08274a8b234c.png)


The next step we need to take is to update travis to run foxy not [noetic](https://github.com/ros-planning/moveit2_tutorials/blob/2dabade2bc9cf56b7b940e7eaeb79813c7afb536/.travis.yml#L13). Is that something you would like me to tackle in this PR or open a new one (it will depend on this one if we do).

MarqRazz commented:
Sorry I was wrong travis does not run `catkin build`, it is only running `clang-format` and `catkin_lint`. This setup passes `clang-format` but when I run `catkin_lint` it cant find any packages and returns the following message.
`catkin_lint: no packages to check`
I assume this is because I removed all of the `catkin` requirements from the cmake and package.xml.

henningkayser commented on package.xml:
```diff
@@ -44,30 +44,30 @@
   <run_depend>tf2_geometry_msgs</run_depend>
   <run_depend>xacro</run_depend>
 
   <test_depend>moveit_resources_panda_moveit_config</test_depend>
```
I'm actually a bit confused that we have a test dependency to `moveit_resources_panda_moveit_config` while the run dependency uses `panda_moveit_config`...

MarqRazz replied:
I was confused by this too, but didn't want to remove it. Right now no tests are enabled so we could remove this and expect the person that ports the [test tutorial](https://github.com/ros-planning/moveit2_tutorials/tree/main/doc/tests) to add it back if needed?? Any suggestions on how you would like to move forward here?

henningkayser commented on CMakeLists.txt:
```diff
@@ -1,75 +1,85 @@
 project(moveit2_tutorials)
 
-set(CMAKE_CXX_STANDARD 14)
+if(NOT "${CMAKE_CXX_STANDARD}")
```
This could use the `moveit_package` macro https://github.com/ros-planning/moveit2/blob/main/moveit_common/cmake/moveit_package.cmake

MarqRazz replied:
That is where I grabbed the current if statement from. I will get that added in.

henningkayser reviewed (commented):
Looks all good to me so far. `rviz_visual_tools` could be installed from source for now, but we should still look into getting `moveit_visual_tools` ready. I'll look into a minimal Travis setup.

MarqRazz commented:
> `rviz_visual_tools` could be installed from source for now

How would you like to accomplish that? I can make a `moveit2_tutorials.repos` file but right now we already require all of [moveit2.repos](https://github.com/ros-planning/moveit2/blob/main/moveit2.repos) to build (or would this work with the current binary release of moveit2 for foxy)?

MarqRazz commented:
@henningkayser and @bostoncleek this is ready for another review when you get a chance.

I added a vsc file `moveit2_tutorials.repos` that we can use to pull down this repo and `rviz_visual_tools`. 
I updated the cmake to use `moveit_package()`.
I added a comment to the `run_depend` to update the tutorials to use `moveit_resources_panda_moveit_config` 
I ensured that all required packages are available by running `rosdep install -r --from-paths . --ignore-src --rosdistro foxy` in my worksapce

MarqRazz commented on .travis.yml:
```diff
@@ -1,32 +1,43 @@
-  matrix:
-    - SCRIPT=htmlproofer.sh
-    - TEST="clang-format catkin_lint"
-    - TEST="clang-tidy-fix"
```
Any other tests you want to run? Should we run clang-tidy-fix?

auman66 replied:
We need to find a way to keep htmlproofer in there

MarqRazz replied:
That test  is included and passes. See lines 25 and 26 of the updated travis file. Also double check my message below on needing to update the script.

MarqRazz commented on .travis.yml:
```diff
@@ -1,32 +1,43 @@
+    - name: "[Foxy] - clang-format, ament_lint"
+      env: TEST="clang-format ament_lint"
+    - name: "[Foxy] - code-coverage"
+      env: TEST=code-coverage
```
Should we remove code-coverage?

henningkayser replied:
might be a good idea :+1:

MarqRazz commented:
[htmlproofer.sh](https://github.com/MarqRazz/moveit2_tutorials/blob/9b118e052f7ce938f3c2d54b5685e7fb2fbce3cb/htmlproofer.sh#L11) test is passing but does it need to be updated? It depends `python-catkin-tools`?

MarqRazz commented on .travis.yml:
```diff
@@ -1,32 +1,43 @@
+    - MOVEIT_CI_TRAVIS_TIMEOUT=85  # Travis grants us 90 min, but we add a safety margin of 5 min
+    - ROS_DISTRO=foxy
+    - ROS_REPO=ros
+    - UPSTREAM_WORKSPACE=moveit2_tutorials.repos
```
So I had to add all of `moveit2.repos` to this file. Is there a way to have 2 repos files here and pull down source from both files?

henningkayser replied:
I think we support multiple repos files, adding multiple comma-separated entries in quotes should work: `"1.repos, 2.repos "`

davetcoleman commented:
I love seeing @MarqRazz already making progress on this!

auman66 reviewed (approved):
@MarqRazz - I still need ROS Melodic to build this locally. Is that correct?

It makes sense to me since there are some Melodic rosdoc_lite and Python 2 stuff, but wanted to check. If this is correct, then its good to go!

griswaldbrooks commented:
@MarqRazz can you rebase? This branch still has the
https://github.com/MarqRazz/moveit2_tutorials/blob/pr-update_cmake/_themes/sphinx_rtd_theme/breadcrumbs.html#L23
that you fixed in https://github.com/ros-planning/moveit2_tutorials/pull/45

MarqRazz commented:
> @MarqRazz - I still need ROS Melodic to build this locally. Is that correct?
> 
> It makes sense to me since there are some Melodic rosdoc_lite and Python 2 stuff, but wanted to check. If this is correct, then its good to go!

Yes @auman66 you still need ROS Melodic to build the html pages locally

auman66 commented:
CI failing is a docker/travis issue. I'll try rerunning and siff it it works out

MarqRazz commented:
Thanks Alex, sorry it took me a while to get back to this, its been a crazy week. I re-added your changes from #47 and after this passes CI I think it is ready to go.

griswaldbrooks commented:
> > @MarqRazz - I still need ROS Melodic to build this locally. Is that correct?
> > It makes sense to me since there are some Melodic rosdoc_lite and Python 2 stuff, but wanted to check. If this is correct, then its good to go!
> 
> Yes @auman66 you still need ROS Melodic to build the html pages locally

@MarqRazz I just spun up a noetic container and with your `viewkeys` fix `build_locally.sh` works
